src/box/iproto.cc

```diff
@@ -1664,6 +1697,14 @@ tx_process_call(struct cmsg *m)
 	tx_reply_error(msg);
 }
 
+static void
+tx_process_call_guarded(struct cmsg *m)
+{
+	tx_start_request(m);
```

Comment:
It's not work for requests which executes in fibers!
Server:
```lua
#!/PATH_TO_TARANTOOL
os.execute("rm -r *.snap *.xlog")
box.cfg {
        listen = "/tmp/1"
}
box.schema.user.grant('guest','execute,write,read','universe')
box.ctl.set_on_shutdown_timeout(10)
fiber = require('fiber')
function simple() fiber.create(function()
                while true do
                        print("XXX")
                        fiber.sleep(1)
                end
        end)
end
require('console').start()
```
Client:
```lua
#!/PATH_TO_TARANTOOL
os.execute("rm -rf *.snap *.xlog")
net_box = require('net.box')
graceful_con = net_box.connect("/tmp/1")
graceful_con.space._session_settings:update('graceful_shutdown', {{'=', 'value', true}})
net_box.set_shutdown_handler(function() graceful_con:shutdown({is_async=true}) end)
graceful_con:call('simple')
require('console').start()
```
Then if you stop server using Ctrl + C it's stopped immediately, but  really function `simple` which was called by client is not still finished!

Reply:
Function `simple` just creates fiber and doesn't wait for it to finish, so in this case `simple` is finished, but fiber is still running and I think, that we don't need to wait this fiber.

Reply:
This is a simple example. What if this function does some work (for example, updates multiple tuples in a fiber, and then asynchronously notify client). I think we need some mechanism based on callbacks, which will allow the user function to notify about the end of the request execution. @Korablev77 what do you think about it?

Reply:
I believe that any working fibers should be forced to finish by shutdown timeout. Still we can send error response to the client that function hasn't been finished in a normal mode and was terminated due to timeout.

Reply:
We have  long requests with `box.session.push` and `is_async` option for sending something from server to client. I think, `graceful shutdown` works with `is_async` option and we decided to do not implement `graceful shutdown` support for `box.session.push` in this patchset.